### PR TITLE
Refresh Dynamic Config page

### PR DESCRIPTION
**Pull request by kbatuigas**
# Description
Additional info on how dynamic config works.

# Reasons
Prompted by [this issue](https://github.com/circleci/circleci-docs/issues/5621), which calls out that the page doesn't mention how pipeline parameters work with the continuation config.

### CODE REVIEW

File: jekyll/_cci2/dynamic-config.md [hunk 1]
```diff
@@ -6,48 +6,48 @@ version:
...
+6. At the end of the `setup` workflow, a `continue` job from the [`continuation` orb](https://circleci.com/developer/orbs/orb/circleci/continuation) must be called. **NOTE:** This does _not_ apply if you want to conditionally execute workflows or steps based on **updates to specified files**, as described in the [Configuration Cookbook]({{ site.baseurl }}/2.0/configuration-cookbook/?section=examples-and-guides#execute-specific-workflows-or-steps-based-on-which-files-are-modified) example.
 
-Now, your project has the ability to dynamically generate and update configuration.
+## How dynamic config works
+{: #how-dynamic-config-works }
 
-Note: While the steps above will make the feature available, your static `config.yml` will continue to work as normal. This feature will not be used until you add the key `setup` with a value of `true` to that `config.yml`.
+CircleCI's dynamic configuration feature uses setup workflows. A setup workflow can contain jobs that compute pipeline parameters, which can then be passed into an additional configuration that potentially exists in other directories. A setup workflow can also be used to generate follow-up pipelines via pre-existing scripts.
```
Comment (sulami) [inline jekyll/_cci2/dynamic-config.md:38]: I wouldn't talk about follow-up pipelines, they're a different and confusing concept in this context.

Comment (sulami) [reply jekyll/_cci2/dynamic-config.md:38]: On the other hand it might be worth mentioning that not only pipeline parameters can be generated, the entire config can be generated within the setup workflow.

File: jekyll/_cci2/dynamic-config.md [hunk 2]
```diff
@@ -6,48 +6,48 @@ version:
 - Cloud
 ---
 
-You may find that instead of manually creating each and every individual CircleCI configuration per project, you would prefer to generate these configurations dynamically, depending on specific [pipeline]({{ site.baseurl }}/2.0/concepts/#pipelines)
-parameters or file-paths.
+Instead of manually creating an individual CircleCI configuration per project, you might prefer in some cases to generate these configurations dynamically, depending on specific [pipeline parameters]({{ site.baseurl }}/2.0/pipeline-variables/) or file paths.
```
[the remark this answers is not in the document]
Comment (kbatuigas) [reply jekyll/_cci2/dynamic-config.md:9]: @sulami To make sure I'm following - is there a specific reason for using "values" instead of "parameters"? For instance I noticed that this [Discuss post](https://discuss.circleci.com/t/intro-to-dynamic-config-via-setup-workflows/39868) refers to pipeline parameters. It does look like either pipeline parameters or values could be used... would "pipeline variables" be a better option here?

File: jekyll/_cci2/dynamic-config.md [hunk 3]
```diff
@@ -6,48 +6,48 @@ version:
...
 
 - Execute conditional workflows/commands
 - Pass pipeline parameter values and/or generate additional configuration
 - Trigger separate `config.yml` configurations which exist outside the default parent `.circleci/` directory
 
-To use our dynamic configuration feature, you can add the key `setup` with a value of `true` to the top-level of your parent configuration file (in the `.circleci/` directory). This will designate that `config.yaml` as a `setup workflow` configuration, enabling you and your team to get up and running with dynamic configuration.
-
-See the [Getting started](#getting-started-with-dynamic-config-in-circleci) section below for more 
-information.
+This becomes particularly useful in cases where your team is using a monorepo, or a single repository, as opposed to using multiple repositories to store your code. In the case of using a monorepo, it would be optimal to only trigger certain builds in specific areas of your project. Otherwise, all of your microservices/sub-projects would go through the entirety of your build, test, and deployment processes when any single update is introduced.
```
[the remark this answers is not in the document]
Comment (kbatuigas) [reply jekyll/_cci2/dynamic-config.md:17]: @sulami Thanks - agreed, this does seem like path filtering is the only scenario involved. My next commit is going to attempt to point out path filtering as a specific use case.

File: jekyll/_cci2/dynamic-config.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sulami) [reply jekyll/_cci2/dynamic-config.md:9]: Pipeline parameters are declared and set by the user, pipeline values are implicitly declared and set by us. Technically there's no term encompassing both, though they can be used interchangeably everywhere. I'm okay with calling the union of those two sets pipeline variables.

File: jekyll/_cci2/dynamic-config.md [hunk 4]
```diff
@@ -4,50 +4,55 @@ title: "Dynamic Configuration"
...
-In both of these (and many other) use cases, automatic, dynamic generation of your configuration files will optimize your CircleCI experience and save your team both time and money.
-
-CircleCI's dynamic configuration feature uses a `setup workflow` configuration. A `setup workflow` can contain jobs that `setup` children pipelines through computed pipeline parameters, or by generating follow-up pipelines via pre-existing scripts. These computed pipeline parameters and/or generated `config.yaml` files can then be passed into an additional `config.yaml` that potentially exists in outside directories.
-
-In summary, CircleCI's dynamic configuration allows you to:
+CircleCI's dynamic configuration feature allows you to:
 
 - Execute conditional workflows/commands
 - Pass pipeline parameter values and/or generate additional configuration
 - Trigger separate `config.yml` configurations which exist outside the default parent `.circleci/` directory
```
Comment (rosieyohannan) [inline jekyll/_cci2/dynamic-config.md:16]:
```suggestion
- Execute conditional workflows/commands.
- Pass pipeline parameter values and/or generate additional configuration.
- Trigger separate `config.yml` configurations, which exist outside the default parent `.circleci/` directory.
```

File: jekyll/_cci2/dynamic-config.md [hunk 5]
```diff
@@ -4,50 +4,55 @@ title: "Dynamic Configuration"
...
+This becomes particularly helpful in cases where your team is using a monorepo, or a single repository, as opposed to using multiple repositories to store your code.
 
-See the [Getting started](#getting-started-with-dynamic-config-in-circleci) section below for more 
-information.
+For instance, you may want to use dynamic configuration to trigger certain workflows based on which files or subdirectories have changed in your repository. Otherwise, all of your microservices/sub-projects would go through the entirety of your build, test, and deployment processes when any single update is introduced. This scenario is also referred to as _path filtering_.
+
+Another scenario in which you could use dynamic configuration is when your project consists multiple modules, where each module requires a separate job. These jobs could be stored in multiple files as _fragments_ of a configuration. Instead of using one full `.circleci/config.yml` file, you could use dynamic configuration to combine these individual fragments into a full configuration when a pipeline is triggered. This scenario can also be referred to as _config splitting_.
```
Comment (rosieyohannan) [inline jekyll/_cci2/dynamic-config.md:22]:
```suggestion
Another scenario in which you could use dynamic configuration is when your project consists of multiple modules, where each module requires a separate job. These jobs could be stored in multiple files as _fragments_ of a configuration. Instead of using one full `.circleci/config.yml` file, you could use dynamic configuration to combine these individual fragments into a full configuration when a pipeline is triggered. This scenario can also be referred to as _config splitting_.
```

File: jekyll/_cci2/dynamic-config.md [hunk 6]
```diff
@@ -4,50 +4,55 @@ title: "Dynamic Configuration"
...
 
-Now, your project has the ability to dynamically generate and update configuration.
+## How dynamic config works
+{: #how-dynamic-config-works }
 
-Note: While the steps above will make the feature available, your static `config.yml` will continue to work as normal. This feature will not be used until you add the key `setup` with a value of `true` to that `config.yml`.
+CircleCI's dynamic configuration feature uses setup workflows. A setup workflow can contain jobs that compute pipeline parameters, which can then be passed into an additional configuration that potentially exists in other directories. A setup workflow can also be used to generate new configurations via pre-existing scripts.
```
Comment (rosieyohannan) [inline jekyll/_cci2/dynamic-config.md:43]: Maybe use italics for "setup workflow" and "continuation" as they have specific meanings?

File: jekyll/_cci2/dynamic-config.md [hunk 7]
```diff
@@ -4,50 +4,55 @@ title: "Dynamic Configuration"
...
+Behind the scenes, the continuation configuration is implemented as a call to a public pipeline continuation API. This API accepts a continuation key, which is a secret, unique-per-pipeline key that is automatically injected into the environment of jobs executed as part of a setup workflow. It also accepts a configuration string, as well as a set of pipeline parameters.
 
-For a basic example on how to use `setup workflows` for dynamic configuration generation, see the [Configuration Cookbook]({{ site.baseurl }}/2.0/configuration-cookbook/?section=examples-and-guides#dynamic-configuration). Included in the cookbook are other more in-depth examples, which will be updated as this feature matures.
+Note that:
+- The setup phase requires `version: 2.1` or higher.
+- A pipeline can only be continued once (i.e. a pipeline cannot be continued with another setup configuration).
```
Comment (rosieyohannan) [inline jekyll/_cci2/dynamic-config.md:49]:
```suggestion
- A pipeline can only be continued once (that is, a pipeline cannot be continued with another setup configuration).
```

File: jekyll/_cci2/dynamic-config.md [hunk 8]
```diff
@@ -4,50 +4,55 @@ title: "Dynamic Configuration"
...
+- A pipeline can only be continued within six hours of its creation.
+- The setup configuration can only include one workflow.
+- Pipeline parameters submitted at continuation time cannot overlap with pipeline parameters submitted at trigger (i.e. setup) time.
```
Comment (rosieyohannan) [inline jekyll/_cci2/dynamic-config.md:52]:
```suggestion
- Pipeline parameters submitted at continuation time cannot overlap with pipeline parameters submitted at trigger (setup) time.
```

File: jekyll/_cci2/dynamic-config.md [hunk 9]
```diff
@@ -57,18 +62,28 @@ For a more in-depth explanation on the behind-the-scenes pipeline creation/conti
...
+**Q:** Can I use a custom executor?
+
+**A:** Custom executors can be used, but require certain dependencies to be installed for the continuation step to work (currently: `curl`, `jq`).
+
+### The continuation orb
 {: #the-continuation-orb }
 
 **Q:** What is the `continuation` orb?
 
 **A:** The `continuation` orb assists CircleCI users in managing the pipeline continuation process easily. The
```
Comment (rosieyohannan) [inline jekyll/_cci2/dynamic-config.md:79]:
```suggestion
**A:** The `continuation` orb assists you in managing the pipeline continuation process. The
```

**Review by rosieyohannan — COMMENTED** [review]
Thanks Kat!

**Review by rosieyohannan — APPROVED** [review]
🎉
